src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml

```diff
@@ -0,0 +1,24 @@
+@using OrchardCore.ContentManagement
+@using OrchardCore.Forms.Models;
+@using OrchardCore.Mvc.Utilities;
+@using Newtonsoft.Json.Linq;
+
+@inject OrchardCore.ContentManagement.Display.IContentItemDisplayManager ContentItemDisplayManager
+
+@{
+    var formPart = ((ContentItem)Model.ContentItem).As<FormPart>();
+    var widgets = (JArray)((JObject)Model.ContentItem.Content)["FlowPart"]["Widgets"];
+    bool hasFileInput = false;
+
+    if(widgets.Count() > 0 )
+    {
+        hasFileInput = widgets?.Where( a=> a["InputPart"] != null && a["InputPart"]["Type"]?.Value<string>() == "file" )?.Count() > 0;   
+    }
+
+    string encType = hasFileInput ? "multipart/form-data" : "application/x-www-form-urlencoded";  
```

Comment:
Should it be an option on the Form part? With a drop-down of predefined values?

Reply:
I thought to make it less technical assign automatically based on widgets added

Reply:
moved to part